js/join.js

```diff
@@ -0,0 +1,290 @@
+  });
+}
+
+// variables for personal details Page
+const city = document.getElementById('city');
+const state = document.getElementById('state');
+const country = document.getElementById('country');
+const next1 = document.getElementById('next1');
+const previous1 = document.getElementById('previous1');
+
+// variables for introduction page
+const introduction = document.getElementById('introduction');
+const skills = document.getElementById('skills');
+const college = document.getElementById('college');
+const forFun = document.getElementById('forFun');
+const funFact = document.getElementById('funFact');
+const next2 = document.getElementById('next2');
+const previous2 = document.getElementById('previous2');
+
+//variables for why RDS Page
+const whyRds = document.getElementById('whyRds');
+const heardAbout = document.getElementById('heardAbout');
+const previous3 = document.getElementById('previous3');
+const previewBtn = document.getElementById('next3');
+
+//variables for preview pages
+const previewFName = document.getElementById('previewFName');
+const previewLName = document.getElementById('previewLName');
+const previewCity = document.getElementById('previewCity');
+const previewState = document.getElementById('previewState');
+const previewCountry = document.getElementById('previewCountry');
+const previewIntro = document.getElementById('previewIntro');
+const previewSkills = document.getElementById('previewSkills');
+const previewInstitution = document.getElementById('previewInstitution');
+const previewFunFact = document.getElementById('previewFunFact');
+const previewForFun = document.getElementById('previewForFun');
+const previewWhyRds = document.getElementById('previewWhyRds');
+const previewHeardAbout = document.getElementById('previewHeardAbout');
+const previous4 = document.getElementById('previous4');
+const submit = document.getElementById('next4');
+
+//Vatiables for Completed page
+const personalLink = document.getElementById('personalLink');
+const copyBtn = document.getElementById('copy');
+
+function fetchSavedDetails() {
+  fetch('https://api.realdevsquad.com/users/self', {
+    headers: { 'content-type': 'application/json' },
+    credentials: 'include',
+  })
+    .then((res) => res.json())
+    .then((res) => {
+      window.localStorage.setItem('firstName', res.first_name);
+      window.localStorage.setItem('lastName', res.last_name);
+      console.log(res);
+      url = `https://api.realdevsquad.com/users/${res.id}/intro`;
+      personalLink.innerText = url;
+    });
+}
+
+const pages = [page1, page2, page3, page4, page5, page6];
+
+function selectPage() {
+  let currentFlowState = window.localStorage.getItem('flowState');
+  showPage(currentFlowState);
+}
+
+function showPage(currentFlowState) {
+  for (let i = 0; i < pages.length; i++) {
+    if (i == currentFlowState) {
+      pages[i].classList.remove('hide-page');
+    } else {
+      pages[i].classList.add('hide-page');
+    }
+  }
```

Comment:
1. We can use `forEach` here
2. If else conditions can be simplified with the toggle property of classList
https://developer.mozilla.org/en-US/docs/Web/API/Element/classList

Reply:
I tried with toggle but it won't work. toggling will not hide the previous page in this scenario as I am using pages[I] for that I will have to write pages[i-1] to toggle on that